tcms/core/middleware.py

```diff
@@ -14,7 +15,11 @@
 class CheckSettingsMiddleware(MiddlewareMixin):
     def process_request(self, request):
         doc_url = "https://kiwitcms.readthedocs.io/en/latest/admin.html#configure-kiwi-s-base-url"
-        site = Site.objects.get(pk=settings.SITE_ID)
+        try:
+            site = Site.objects.get(pk=settings.SITE_ID)
+        except:
```

Comment:
this needs a specific exception class. Could be something around DB connections or it could be ObjectDoesNotExist if there is a DB connection but migrations (possibly not all) have not been applied. 

From what I can see it is `django.db.utils.OperationalError` with message `no such table: django_site` (hint this could probably happen also if the DB server suddenly goes south in an otherwise perfectly configured system).

Reply:
Also needs to be its own middleware sitting as the first one in the chain.